### PR TITLE
Add `use_asan` build option

### PR DESCRIPTION
**Pull request by RandomShaper**
Works with https://github.com/godotengine/godot/pull/93152.

### CODE REVIEW

File: SConstruct [hunk 1]
```diff
@@ -78,6 +78,8 @@ opts.Add(
     )
 )
 
+opts.Add(BoolVariable("use_asan", "Use address sanitizer (ASAN)", False))
```
Comment (bruvzg) [inline SConstruct]: This should be marked as Windows and MSVC only (or support other platforms as well).

File: SConstruct [hunk 2]
```diff
@@ -177,6 +179,12 @@ if env["platform"] == "macos":
 elif env["platform"] == "windows":
     env.AppendUnique(CPPDEFINES=["WINVER=0x0603", "_WIN32_WINNT=0x0603"])
 
+# Sanitizers.
+if env.get("use_asan", False):
```
Comment (bruvzg) [inline SConstruct]:
```suggestion
if env.get("use_asan", False) and env.get("is_msvc", False):
```

File: SConstruct (hunk 1 above)
Comment (RandomShaper) [reply SConstruct]: I thought these changes already support GCC/Clang. The main build system in Godot does the same: add the `/INFERSANLIBS` that non-MSVC will ignore and also `/fsanitize=address` that both MSVC and others will honor.

I can be deeply wrong, though.

Comment (bruvzg) [reply SConstruct]: Not, both of these are MSVC only, and only defined in the MSVC code path of the main repo. For GCC/Clang it's:

```
env.Append(CCFLAGS=["-fsanitize=address,pointer-subtract,pointer-compare"])
env.Append(LINKFLAGS=["-fsanitize=address"])
```

Comment (bruvzg) [reply SConstruct]: For MinGW it seems like it's not used in the main repo at all (only for Linux, macOS, and Web), not sure if it's not supported or just missing.

Comment (RandomShaper) [reply SConstruct]: I see. I will then just keep all this for MSVC for now.

**Review by bruvzg — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**